model/src/main/java/com/oasis/model/EmployeeModel.java

```diff
@@ -0,0 +1,51 @@
+package com.oasis.model;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+import org.bson.BsonTimestamp;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import java.util.Date;
+
+import static com.oasis.model.CollectionNames.EMPLOYEE_COLLECTION_NAME;
+
+@Document(collection = EMPLOYEE_COLLECTION_NAME)
+@Data
+public class EmployeeModel {
+    @Id
+    @Setter(AccessLevel.PRIVATE)
+    private String _id;
+
+    @Field("employeeFullname")
+    private String fullname;
+
+    @Field("employeeDOB")
+    private Date dob;
+
+    @Field("employeeUsername")
+    private String username;
+
+    @Field("employeePassword")
+    private String password;
+
+    @Field("employeePhone")
+    private String phone;
+
+    @Field("employeeJobTitle")
+    private String jobTitle;
+
+    @Field("employeeDivision")
+    private String division;
+
+    @Field("employeeSupervisingCount")
+    private Integer supervisingCount;
+
+    private String supervisionId;
+    private BsonTimestamp createdDate;
```

Comment:
move id, createdDate, updatedDate, createdBy, updatedBy to BaseEntity and make this class to extends BaseEntity

Reply:
Commit ID: 9465eb3